operator/iamkeycontroller/pipeline.go

```diff
@@ -61,11 +63,17 @@ func NewPipeline(client client.Client, recorder event.Recorder, exoscaleClient *
 	}
 }
 
-func toConnectionDetails(iamKey *exoscalesdk.IAMAccessKey) managed.ConnectionDetails {
+func toConnectionDetails(iamKey *exoscalesdk.IAMAccessKey) (managed.ConnectionDetails, error) {
+	if iamKey.Key == nil {
+		return nil, fmt.Errorf("iamKey key not found in connection details")
```

Comment:
```suggestion
		return nil, errors.New("iamKey key not found in connection details")
```

`fmt.Errorf` is used when you need a format string in the error which gets replaced by something (usually `%w` which wraps an error). If you just need a normal error value, `errors.New()` is the way to go.

Reply:
true that, cheers